src/public/media.ts

```diff
@@ -350,7 +350,25 @@ export function selectMedia(mediaInputs: MediaInputs, callback: (error: SdkError
 
   const params = [mediaInputs];
   const messageId = sendMessageRequestToParent('selectMedia', params);
-  GlobalVars.callbacks[messageId] = callback;
+
+  // What comes back from native at attachments would just be objects and will be missing getMedia method on them.
+  GlobalVars.callbacks[messageId] = (err: SdkError, attachs: Media[]) => {
+    let mediaArray: Media[] = null;
+    if (attachs) {
+      mediaArray = [];
+      for (let attachment of attachs) {
+        const media = new Media();
```

Comment:
Why not add a constructor that just takes an attachment? You could even store that internally rather than copying each prop one by one.

Reply:
Yes. It makes sense. I moved it into a constructor. Unfortunately I cannot use Object.assign as we are targeting es5. So had to copy properties one by one. Please let me know if there is a better way to do this.